content/docs/grammar-language.md

```diff
@@ -170,82 +76,68 @@ Person returns OtherType:
 ```
 The parser rule `Person` will now lead to the creation of objects of type `OtherType` instead of `Person`.
 
-#### Keywords
-Keywords are *inline terminals* which need to match the terminal `STRING` as defined by he Langium grammar. This means that a keyword can be a sequence of one or many arbitrary characters surrounded by single or double quotes.
-Keywords are meant to guide the parser in defining what type of object needs to be parsed.
-
-Consider the following:
-```
-Student:
-    name=ID;
-
-Teacher:
-    name=ID;
-
-Person:
-    Student | Teacher;
+### The Entry Rule
+The *entry rule* is a parser rule that defines the starting point of the parsing step. The *entry rule* starts with the keyword `entry` and matches other parser rules.
 ```
-In this example, a `Person` can either be a `Student` or a `Teacher`. This grammar is ambiguous because the parser rules `Student` and `Teacher` are identical. The parser will not be able to differentiate between the parser rules for `Student` and `Teacher` when trying to parse a `Person`. 
-Keywords can help removing such ambiguity and guide the parser in defining if a `Student` or `Teacher` needs to be parsed.
-We can add a keyword to the parser rule `Student`, `Teacher`, or to both of them:
+entry Model:
+    (persons+=Person | greetings+=Greeting)*;
 ```
-Student:
-    'student' name=ID;
+In this example, the *entry rule* `Model` defines a group of alternatives. The parser will go through the input document and try to parse `Person` or `Greeting` objects and add them to the `persons` or `greetings` arrays, respectively. The parser reads the token stream until all inputs have been consumed.
```

Comment:
Am I correct that the parser reads all inputs until the end of the document **because** of the cardinality * ?

Wouldn't the parser stop after the first match if we do not have the cardinality * ?

If so then I think it would be good to mention it here.

```suggestion
In this example, the *entry rule* `Model` defines a group of alternatives. The parser will go through the input document and try to parse `Person` or `Greeting` objects and add them to the `persons` or `greetings` arrays, respectively. The parser reads the token stream until all inputs have been consumed thanks to the cardinality `*`.
```

Reply:
The parser always tries to consume the whole token stream. In case there are syntax errors, it may stop earlier (depending on the error recovery capabilities), but that's independent of the cardinality.

Reply:
Of course it's a syntax error to write more than one person or greeting if the `* ` is omitted, but the parser will still read the superfluous tokens.